iceoryx2-ffi/ffi/src/api/config.rs

```diff
@@ -12,22 +12,1639 @@
+        .value
+        .global
+        .service
+        .static_config_storage_suffix
+        .as_c_str()
+}
+
+/// Sets the suffix of the static config file
+///
+/// Returns: [`iox2_semantic_string_error_e`](crate::api::iox2_semantic_string_error_e) when an
+/// invalid file name was provided
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+/// * `value` - A valid file name containing the suffix
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_global_service_set_static_config_storage_suffix(
+    handle: iox2_config_ref_h,
+    value: *const c_char,
+) -> c_int {
+    debug_assert!(!handle.is_null());
+
+    let config = &mut *handle.as_type();
+    match FileName::from_c_str(value) {
+        Ok(n) => {
+            config
+                .value
+                .as_mut()
+                .value
+                .global
+                .service
+                .static_config_storage_suffix = n;
+            IOX2_OK as _
+        }
+        Err(e) => e as c_int,
+    }
+}
+
+/// Returns the suffix of the dynamic config file
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_global_service_dynamic_config_storage_suffix(
+    handle: iox2_config_ref_h,
+) -> *const c_char {
+    debug_assert!(!handle.is_null());
+
+    let config = &*handle.as_type();
+    config
+        .value
+        .as_ref()
+        .value
+        .global
+        .service
+        .dynamic_config_storage_suffix
+        .as_c_str()
+}
+
+/// Sets the suffix of the dynamic config file
+///
+/// Returns: [`iox2_semantic_string_error_e`](crate::api::iox2_semantic_string_error_e) when an
+/// invalid file name was provided
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+/// * `value` - A valid file name containing the suffix
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_global_service_set_dynamic_config_storage_suffix(
+    handle: iox2_config_ref_h,
+    value: *const c_char,
+) -> c_int {
+    debug_assert!(!handle.is_null());
+
+    let config = &mut *handle.as_type();
+    match FileName::from_c_str(value) {
+        Ok(n) => {
+            config
+                .value
+                .as_mut()
+                .value
+                .global
+                .service
+                .dynamic_config_storage_suffix = n;
+            IOX2_OK as _
+        }
+        Err(e) => e as c_int,
+    }
+}
+
+/// Returns the second part of the time of how long another process will wait until the service
+/// creation is finalized
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_global_service_creation_timeout_sec(
+    handle: iox2_config_ref_h,
+) -> u64 {
```

Comment:
Have you thought about using a `timespec` here?

Reply:
Yes, but timespec is only available on posix platforms, not windows. I could of course introduce an `iox_timespec` and return this thing here or use two pointers to u64 and u32 ... I think I will adjust this to the pointer approach.